src/main/java/mate/academy/rickandmorty/service/character/CharacterServiceImpl.java

```diff
@@ -0,0 +1,52 @@
+package mate.academy.rickandmorty.service.character;
+
+import java.util.List;
+import java.util.Random;
+import lombok.RequiredArgsConstructor;
+import mate.academy.rickandmorty.dto.CharacterDto;
+import mate.academy.rickandmorty.dto.CharacterSearchParameters;
+import mate.academy.rickandmorty.mapper.CharacterMapper;
+import mate.academy.rickandmorty.model.Character;
+import mate.academy.rickandmorty.repository.character.CharacterRepository;
+import mate.academy.rickandmorty.repository.character.CharacterSpecificationBuilder;
+import mate.academy.rickandmorty.service.CharacterClient;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class CharacterServiceImpl implements CharacterService {
+    public static final int ORIGIN = 1;
+    private final static Random random = new Random();
+    private final CharacterRepository characterRepository;
+    private final CharacterSpecificationBuilder specificationBuilder;
+    private final CharacterMapper characterMapper;
+    private final CharacterClient characterClient;
+
+    @Override
+    public List<CharacterDto> search(CharacterSearchParameters searchParams) {
+        fillDataBase();
+        Specification<Character> characterSpecification = specificationBuilder.build(searchParams);
+        return characterRepository.findAll(characterSpecification)
+                .stream()
+                .map(characterMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public CharacterDto getRandomCharacter() {
+        fillDataBase();

```

Comment:
this should be executed only on application startup. google about options how to implement that
```suggestion
        Specification<Character> characterSpecification = specificationBuilder.build(searchParams);
        return characterRepository.findAll(characterSpecification)
                .stream()
                .map(characterMapper::toDto)
                .toList();
    }

    @Override
    public CharacterDto getRandomCharacter() {
```